### PR TITLE
Make dynamic worker selection actually work as intended

### PR DESCRIPTION
**Pull request by fjy**
This feature is a pain to unit test, as with all dynamic configuration changes, and testing these features should belong with integration tests.

I tested this locally with mysql and an indexing service running in remote mode.

This PR also cleans up deprecated code no longer needed for 0.7.0

### CODE REVIEW

File: indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordResource.java [hunk 1]
```diff
@@ -198,37 +192,6 @@ public Response setWorkerConfig(
     return Response.ok().build();
   }
 
-
-  @Deprecated
-  @GET
-  @Path("/worker/setup")
```
Comment (drcrallen) [inline indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordResource.java]: Please add to release notes even if it was previously marked Deprecated

File: services/src/main/java/io/druid/cli/CliOverlord.java [hunk 2]
```diff
@@ -200,7 +199,6 @@ private void configureRunners(Binder binder)
             biddy.addBinding("remote").to(RemoteTaskRunnerFactory.class).in(LazySingleton.class);
             binder.bind(RemoteTaskRunnerFactory.class).in(LazySingleton.class);
 
-            JacksonConfigProvider.bind(binder, WorkerSetupData.CONFIG_KEY, WorkerSetupData.class, null);
```
Comment (drcrallen) [inline services/src/main/java/io/druid/cli/CliOverlord.java]: Same, can this be noted in the release notes?

Comment (drcrallen) [inline services/src/main/java/io/druid/cli/CliOverlord.java]: er, yep, meant to keep this comment:

Can this be noted in the release notes as well?

**Comment by xvrl** [comment]
squash commits please

**Pull request merged**